File: nestjs/08-nest-continent-dto/src/modules/continent/continent.controller.ts
import {
  Body,
  Controller,
  Delete,
  Get,
  Param,
  ParseIntPipe,
  Post,
  Put,
} from '@nestjs/common';
import { ContinentService } from './continent.service';
import { CreateContinentDto, UpdateContinentDto } from './continent.dto';

@Controller('continents')
export class ContinentController {
  constructor(private readonly continentService: ContinentService) {}

  @Get()
  getAll() {
    return this.continentService.getAll();
  }

  @Get(':id')
  getById(@Param('id', ParseIntPipe) id: number) {
    return this.continentService.getById(id);
  }

  @Post()
  create(@Body() dto: CreateContinentDto) {
    return this.continentService.create(dto);
  }

  @Put(':id')
  update(
    @Param('id', ParseIntPipe) id: number,
    @Body() dto: UpdateContinentDto,
  ) {
    return this.continentService.update(id, dto);
  }

  @Delete(':id')
  delete(@Param('id', ParseIntPipe) id: number) {
    return this.continentService.delete(id);
  }
}

File: springboot-java-25/07-springboot-repository-jpa/src/main/java/com/ganatan/starter/continents/ContinentController.java
package com.ganatan.starter.continents;

import java.util.List;

import com.ganatan.starter.continents.dto.ContinentRequestDTO;
import com.ganatan.starter.continents.dto.ContinentResponseDTO;
import jakarta.validation.Valid;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;

@RestController
@RequestMapping("/continents")
public class ContinentController {

  private final ContinentService service;

  public ContinentController(ContinentService service) {
    this.service = service;
  }

  @GetMapping
  public List<ContinentResponseDTO> getAllContinents() {
    return service.getAll().stream()
      .map(this::toResponse)
      .toList();
  }

  @GetMapping("/{id}")
  public ContinentResponseDTO getContinentById(@PathVariable int id) {
    return toResponse(service.getById(id));
  }

  @PostMapping
  @ResponseStatus(HttpStatus.CREATED)
  public ContinentResponseDTO createContinent(@Valid @RequestBody ContinentRequestDTO body) {
    return toResponse(service.create(body.name()));
  }

  @PutMapping("/{id}")
  public ContinentResponseDTO updateContinent(
    @PathVariable int id,
    @Valid @RequestBody ContinentRequestDTO body
  ) {
    return toResponse(service.update(id, body.name()));
  }

  @DeleteMapping("/{id}")
  @ResponseStatus(HttpStatus.NO_CONTENT)
  public void deleteContinent(@PathVariable int id) {
    service.delete(id);
  }

  private ContinentResponseDTO toResponse(Continent continent) {
    return new ContinentResponseDTO(
      continent.id(),
      continent.name()
    );
  }

}



//package com.ganatan.starter.continents;
//
//import java.util.List;
//
//import com.ganatan.starter.continents.dto.ContinentRequestDTO;
//import jakarta.validation.Valid;
//import org.springframework.http.HttpStatus;
//import org.springframework.web.bind.annotation.*;
//
//@RestController
//@RequestMapping("/continents")
//public class ContinentController {
//
//  private final ContinentService service;
//
//  public ContinentController(ContinentService service) {
//    this.service = service;
//  }
//
//  @GetMapping
//  public List<Continent> getAllContinents() {
//    return service.getAll();
//  }
//
//  @GetMapping("/{id}")
//  public Continent getContinentById(@PathVariable int id) {
//    return service.getById(id);
//  }
//
//  @PostMapping
//  @ResponseStatus(HttpStatus.CREATED)
//  public Continent createContinent(@Valid @RequestBody ContinentRequestDTO body) {
//
//    return service.create(body.name());
//  }
//
//  @PutMapping("/{id}")
//  public Continent updateContinent(
//    @PathVariable int id,
//    @Valid @RequestBody ContinentRequestDTO body
//  ) {
//    return service.update(id, body.name());
//  }
//
//  @DeleteMapping("/{id}")
//  @ResponseStatus(HttpStatus.NO_CONTENT)
//  public void deleteContinent(@PathVariable int id) {
//    service.delete(id);
//  }
//}
